### PR TITLE
Fix learning rate scheduler performance issue

### LINKED ISSUE #12260: Learning rate scheduler performance issue
In our current learning rate scheduler implementation, the `learning rate` is a CPU scalar variable. It results in a problem that every time a parameter is updated, Fluid has to launch a kernel to copy the `learning rate` scalar to GPU. It is extremely inefficient.

### PR DESCRIPTION
**Pull request by JiayiFeng**
fixes #12260 

ResNet50 with piecewise_decay
time cost of `exe.run()`:

|               | 1 GPU |  4 GPU  |  8 GPU |
|--------|--------|---------|---------| 
|before  | 0.75s   |   1.41s    |  1.52s     |
|after     | 0.75s   |   0.88s   |  0.96s    |

### CODE REVIEW

**Comment by jacquesqiao** [comment]
有多少性能提升？可以贴一些数据

**Comment by JiayiFeng** [comment]
恩，这块还在调，稳定后会发一个在ResNet50上的结果

File: python/paddle/fluid/tests/unittests/test_learning_rate_scheduler.py [hunk 1]
```diff
@@ -91,20 +91,21 @@ def check_decay(self, python_decay_fn, fluid_decay_fn, kwargs):
...
+        with fluid.program_guard(main_prog, startup_prog):
+            decayed_lr = fluid_decay_fn(**kwargs)
 
         place = fluid.CPUPlace()
         exe = fluid.Executor(place)
 
-        exe.run(fluid.default_startup_program())
+        exe.run(startup_prog)
 
-        fluid.memory_optimize(fluid.default_main_program())
```
Comment (jacquesqiao) [inline python/paddle/fluid/tests/unittests/test_learning_rate_scheduler.py:102]: 这里为啥注释掉？有bug？

Comment (JiayiFeng) [reply python/paddle/fluid/tests/unittests/test_learning_rate_scheduler.py:102]: 本来会触发一个显存优化的bug，现在这个bug已经修复，这里可以加回去了

**Review by jacquesqiao — APPROVED** [review]
LGTM!
